packages/config/src/class-config.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * The base Jetpack configuration class file.
+ *
+ * @package automattic/jetpack-config
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\Sync\Actions as Sync_Actions;
+use Automattic\Jetpack\Sync\Main as Sync_Main;
+use Automattic\Jetpack\Plugin\Tracking as Plugin_Tracking;
+use Automattic\Jetpack\Terms_Of_Service;
+
+/**
+ * The configuration class.
+ */
+class Config {
```

Comment:
This seems like a really generic class name. Can we be more specific?

 How would another plugin use this config class?

I think this class seems really Jetpack specific so in that sense I don't think we need to have it here at a package at all. Unless there is a reason for having this inside that I don't quite follow. 

I think having this just be part of separate class in Jetpack that just gets loaded as jetpack gets loaded might work better. Then we also avoid this config package not being able to be loaded from another plugin. Since you would need to call it and then it would get autoloaded before all the plugins have been loaded. Since it calls `add_action('plugins_loaded',... `

Sorry I am late to review this. :(

Reply:
Thanks for getting to it, I'm happy to get as many opinions on this as I can!

The idea behind using it as a package is to have a means to declare Jetpack package features that you're going to be using. So as a package consumer you install all the packages and do something like this:

```
new Config();
$config->ensure( 'jetpack_sync' );
```

This code needs to be run at `plugins_loaded` at highest priority, so that all actions that are added here would be run subsequently. This way we're making sure that we're loading all plugins before we even start autoloading things, even the Config class will be properly autoloaded.

I just realized that we have another problem - the packages will be initialized twice if they are enabled by two different package consumers. I'll work on that.

Oh, by the way, there's a full example in the package README file, you can check that too.

Reply:
Thinking ahead, I'd like to slowly convert our existing modules to packages, so this mech would be what loads up everything eventually (goes to the future iteration where this package is able to enable things more dynamically).

I'm okay with the `Config` name, but devil's advocate, would `Init` or `Initialize` be a better package name/class? If all this package does is initializes Jetpack features based on passed args, maybe that would be better?